safesql/sqlwrap.go

```diff
@@ -21,139 +21,276 @@ import (
+func (db DB) Close() error {
+	return db.db.Close()
+}
+
+// Conn is a tyny wrapper for https://pkg.go.dev/sql#DB.Conn
 func (db DB) Conn(ctx context.Context) (Conn, error) {
 	c, err := db.db.Conn(ctx)
 	return Conn{c}, err
 }
+
+// Exec is a tyny wrapper for https://pkg.go.dev/sql#DB.Exec
 func (db DB) Exec(query TrustedSQLString, args ...interface{}) (Result, error) {
 	return db.db.Exec(query.s, args)
 }
+
+// ExecContext is a tyny wrapper for https://pkg.go.dev/sql#DB.ExecContext
 func (db DB) ExecContext(ctx context.Context, query TrustedSQLString, args ...interface{}) (Result, error) {
 	return db.db.ExecContext(ctx, query.s, args)
 }
-func (db DB) Ping() error                                   { return db.db.Ping() }
-func (db DB) PingContext(ctx context.Context) error         { return db.db.PingContext(ctx) }
-func (db DB) Prepare(query TrustedSQLString) (*Stmt, error) { return db.db.Prepare(query.s) }
+
+// Ping is a tyny wrapper for https://pkg.go.dev/sql#DB.Ping
+func (db DB) Ping() error {
+	return db.db.Ping()
+}
+
+// PingContext is a tyny wrapper for https://pkg.go.dev/sql#DB.PingContext
+func (db DB) PingContext(ctx context.Context) error {
+	return db.db.PingContext(ctx)
+}
+
+// Prepare is a tyny wrapper for https://pkg.go.dev/sql#DB.Prepare
+func (db DB) Prepare(query TrustedSQLString) (*Stmt, error) {
+	return db.db.Prepare(query.s)
+}
+
+// PrepareContext is a tyny wrapper for https://pkg.go.dev/sql#DB.PrepareContext
 func (db DB) PrepareContext(ctx context.Context, query TrustedSQLString) (*Stmt, error) {
 	return db.db.PrepareContext(ctx, query.s)
 }
+
+// Query is a tyny wrapper for https://pkg.go.dev/sql#DB.Query
 func (db DB) Query(query TrustedSQLString, args ...interface{}) (*Rows, error) {
 	return db.db.Query(query.s, args)
 }
+
+// QueryContext is a tyny wrapper for https://pkg.go.dev/sql#DB.QueryContext
 func (db DB) QueryContext(ctx context.Context, query TrustedSQLString, args ...interface{}) (*Rows, error) {
 	return db.db.QueryContext(ctx, query.s, args)
 }
+
+// QueryRow is a tyny wrapper for https://pkg.go.dev/sql#DB.QueryRow
 func (db DB) QueryRow(query TrustedSQLString, args ...interface{}) *Row {
 	return db.db.QueryRow(query.s, args)
 }
+
+// QueryRowContext is a tyny wrapper for https://pkg.go.dev/sql#DB.QueryRowContext
 func (db DB) QueryRowContext(ctx context.Context, query TrustedSQLString, args ...interface{}) *Row {
 	return db.db.QueryRowContext(ctx, query.s, args)
 }
 
-// Go 1.15 func (db DB) SetConnMaxIdleTime(d time.Duration) { db.db.SetConnMaxIdleTime(d) }
-func (db DB) SetConnMaxLifetime(d time.Duration) { db.db.SetConnMaxLifetime(d) }
-func (db DB) SetMaxIdleConns(n int)              { db.db.SetMaxIdleConns(n) }
-func (db DB) SetMaxOpenConns(n int)              { db.db.SetMaxOpenConns(n) }
-func (db DB) Stats() DBStats                     { return db.db.Stats() }
+// Only available in Go 1.15+
+// func (db DB) SetConnMaxIdleTime(d time.Duration) { db.db.SetConnMaxIdleTime(d) }
```

Comment:
why is this commented out?

Reply:
Because otherwise this would only compile with super-recent versions of Go.

I moved it to a file that only gets compiled if go 1.15+ is used.